src/YMenuBar.cc

```diff
@@ -0,0 +1,129 @@
+/*
+  Copyright (c) [2020] SUSE LLC
+  This library is free software; you can redistribute it and/or modify
+  it under the terms of the GNU Lesser General Public License as
+  published by the Free Software Foundation; either version 2.1 of the
+  License, or (at your option) version 3.0 of the License. This library
+  is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+  License for more details. You should have received a copy of the GNU
+  Lesser General Public License along with this library; if not, write
+  to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+  Floor, Boston, MA 02110-1301 USA
+*/
+
+
+/*-/
+
+  File:		YMenuBar.cc
+
+  Author:	Stefan Hundhammer <shundhammer@suse.de>
+
+/-*/
+
+
+#define YUILogComponent "ui"
+#include "YUILog.h"
+
+#include "YUISymbols.h"
+#include "YShortcut.h"
+#include "YMenuBar.h"
+
+
+using std::string;
+
+
+struct YMenuBarPrivate
+{
+    YMenuBarPrivate()
+	{}
+
+    int dummy;
+};
+
+
+
+
+YMenuBar::YMenuBar( YWidget * parent )
+    : YMenuWidget( parent,
+                   "" )         // label
+    , priv( new YMenuBarPrivate() )
+{
+    YUI_CHECK_NEW( priv );
+
+    setStretchable( YD_HORIZ, true );
+    setStretchable( YD_VERT,  false );
+}
+
+
+YMenuBar::~YMenuBar()
+{
+    // NOP
+}
+
+
+YMenuItem *
+YMenuBar::addMenu( const std::string & label,
+                   const std::string & iconName )
+{
+    YMenuItem * menu = new YMenuItem( label, iconName );
+    addItem( menu );
+
+    return menu;
+}
+
+
+const YPropertySet &
+YMenuBar::propertySet()
+{
+    static YPropertySet propSet;
```

Comment:
If I am not wrong, this static `propSet` is shared between all instances of `YMenuBar`. Is this class designed to have only one Menu Bar? Or are that properties shared with all menu bars?

Reply:
Tis is only the _description_ of the properties for use in the UI interpreter. This is always static in all widgets.

Don't confuse this with the actual instance variables that really hold the values that is described by those YPropertySets.